src/OpenSpaceToolkit/Astrodynamics/Flight/Profile/Model/Tabulated.cpp

```diff
@@ -104,35 +104,36 @@ State Tabulated::calculateStateAt(const Instant& anInstant) const
         const Real ratio = Duration::Between(previousState.accessInstant(), anInstant).inSeconds() /
                            Duration::Between(previousState.accessInstant(), nextState.accessInstant()).inSeconds();
 
-        return {
-            anInstant,
-            Position {
-                previousState.getPosition().accessCoordinates() +
-                    ratio *
-                        (nextState.getPosition().accessCoordinates() - previousState.getPosition().accessCoordinates()),
-                previousState.getPosition().getUnit(),
-                previousState.getPosition().accessFrame()
-            },
-            Velocity {
-                previousState.getVelocity().accessCoordinates() +
-                    ratio *
-                        (nextState.getVelocity().accessCoordinates() - previousState.getVelocity().accessCoordinates()),
-                previousState.getVelocity().getUnit(),
-                previousState.getVelocity().accessFrame()
-            },
-            Quaternion::SLERP(previousState.getAttitude(), nextState.getAttitude(), ratio),
-            previousState.getAngularVelocity() +
-                ratio * (nextState.getAngularVelocity() - previousState.getAngularVelocity()),
-            previousState.getFrame()
-        };
+        return State(
+                   anInstant,
+                   Position {
+                       previousState.getPosition().accessCoordinates() +
+                           ratio * (nextState.getPosition().accessCoordinates() -
+                                    previousState.getPosition().accessCoordinates()),
+                       previousState.getPosition().getUnit(),
+                       previousState.getPosition().accessFrame()
+                   },
+                   Velocity {
+                       previousState.getVelocity().accessCoordinates() +
+                           ratio * (nextState.getVelocity().accessCoordinates() -
+                                    previousState.getVelocity().accessCoordinates()),
+                       previousState.getVelocity().getUnit(),
+                       previousState.getVelocity().accessFrame()
+                   },
+                   Quaternion::SLERP(previousState.getAttitude(), nextState.getAttitude(), ratio),
+                   previousState.getAngularVelocity() +
+                       ratio * (nextState.getAngularVelocity() - previousState.getAngularVelocity()),
+                   previousState.getFrame()
+        )
+            .inFrame(Frame::GCRF());
     }
     else if (stateRange.first != nullptr)
     {
-        return *(stateRange.first);
+        return (*(stateRange.first)).inFrame(Frame::GCRF());
```

Comment:
```suggestion
        return State(*(stateRange.first)).inFrame(Frame::GCRF());
```

Reply:
We don't need to create a new state, we're just converting the state that this is pointing to and converting that and returning. Seems unnecessary.